bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveNodeNamingCommandClass.java

```diff
@@ -0,0 +1,360 @@
+	 * @param endpoint the endpoint this Command class belongs to
+	 */
+	public ZWaveNodeNamingCommandClass(ZWaveNode node,
+			ZWaveController controller, ZWaveEndpoint endpoint) {
+		super(node, controller, endpoint);
+	}
+
+
+	public String getName() {
+		if( initialiseName ) {
+			return name;
+		}
+		return null;
+	}
+	
+	public String getLocation() {
+		if( initialiseLocation ) {
+			return location;
+		}
+		return null;
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public CommandClass getCommandClass() {
+		return CommandClass.NODE_NAMING;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void handleApplicationCommandRequest(SerialMessage serialMessage,
+			int offset, int endpoint) {
+		logger.debug("NODE {}: Received NodeNaming Command Class Request", this.getNode().getNodeId());
+		int command = serialMessage.getMessagePayloadByte(offset);
+		switch (command) {
+			
+			case NAME_SET:
+				logger.debug("NODE {}: Name Set sent to the controller will be processed as Name Report", this.getNode().getNodeId());
+				// Process this as if it was a value report.
+				processNameReport(serialMessage, offset, endpoint);
+				initialiseName = true;
+				break;
+				
+			case LOCATION_SET:
+				logger.debug("NODE {}: Location Set sent to the controller will be processed as Location Report", this.getNode().getNodeId());
+				// Process this as if it was a value report.
+				processLocationReport(serialMessage, offset, endpoint);
+				initialiseLocation = true;
+				break;
+				
+			case LOCATION_GET:
+			case NAME_GET:
+				logger.warn(String.format("Command 0x%02X not implemented.", command));
+				return;
+				
+			case NAME_REPORT:
+				logger.trace("NODE {}: Process Name Report", this.getNode().getNodeId());
+				processNameReport(serialMessage, offset, endpoint);
+				initialiseName = true;
+				break;
+				
+			case LOCATION_REPORT:
+				logger.trace("NODE {}: Process Location Report", this.getNode().getNodeId());
+				processLocationReport(serialMessage, offset, endpoint);
+				initialiseLocation = true;
+				break;
+				
+			default:
+				logger.warn(String.format("Unsupported Command 0x%02X for command class %s (0x%02X).", 
+					command, 
+					this.getCommandClass().getLabel(),
+					this.getCommandClass().getKey()));
+		}
+		
+		// If both name and location have been initialized the initialization is done.
+		initialiseDone = initialiseName && initialiseLocation;
+	}
+	
+	/**
+	 * Get a string from the serial message
+	 * @param serialMessage
+	 * @param offset
+	 * @return String
+	 */
+	protected String getString(SerialMessage serialMessage, int offset) { 
+		int charPresentation = serialMessage.getMessagePayloadByte(offset + 1);
+		
+		// First 5 bits are reserved so 0 them
+		charPresentation = 0x07 & charPresentation;
+		
+		switch (charPresentation) {
+			case ENCODING_ASCII:
+				logger.info("NODE {} : Node Name is encoded with standard ASCII codes", this.getNode().getNodeId());
```

Comment:
I think this should be a debug message - most users won't care about seeing this in their log.  Same for the next 3 instances.